packages/core/lib/types/payloadStoreTypes.ts

```diff
@@ -0,0 +1,15 @@
+export interface PayloadStore {
+  /** Store the payload and return a key that can be used to retrieve it later. */
+  storePayload(payload: string): Promise<string>
```

Comment:
would be nice to also support Readable, although in that case we also need to require passing the payload size as well

Reply:
Considering that the payload store is used internally, it does not make much sense to support both types, so I've changed it to be just readable.

Reply:
how will that work in systems that don't support writing to FS? we might need some fallback there

Reply:
Please check this implementation: https://github.com/kibertoad/message-queue-toolkit/pull/167/commits/1fb4c534068227213de5d855442eab802a710c0a. It should provide more flexibility, including avoiding file system-based serialization.